src/librustc_mir/transform/qualify_min_const_fn.rs

```diff
@@ -191,7 +191,12 @@ fn check_rvalue(
             _,
             _,
         ) => Err((span, "function pointer casts are not allowed in const fn".into())),
-        Rvalue::Cast(CastKind::Pointer(PointerCast::Unsize), _, _) => {
+        Rvalue::Cast(CastKind::Pointer(PointerCast::Unsize), op, cast_ty) => if cast_ty.is_slice() {
+            check_operand(tcx, op, span, def_id, body)?;
+            // Casting/coercing things to slices is fine.
+            Ok(())
+        } else {
+            // We just can't allow trait objects until we have figured out trait method calls.
```

Comment:
What about types like `(i32, [i32])`? There are more unsized types than just slices and trait objects.

Reply:
oh heh... oops, fixed.